File: awa/sdk_dist/csharp/Awa.Client/Utilities/Activity/GetDirectoryInfoActivity.cs
using System;
using System.Threading.Tasks;
using Awa.Client.Constants;
using Awa.Client.Models;

namespace Awa.Client.Utilities.Activity
{
    public static partial class ActivityUtilities
    {
        /// <summary>
        /// Get information about a single directory including its immediate files and subdirectories.
        /// </summary>
        /// <param name="directoryPath">The path to the directory to analyze.</param>
        /// <returns>
        /// A FolderInfo object containing:
        /// - path: The directory path
        /// - files: List of file names (not full paths)
        /// - subdirectories: List of subdirectory names (not full paths)
        /// </returns>
        /// <remarks>
        /// This function examines a single directory and returns information about its
        /// immediate contents without recursing into subdirectories. It provides lists
        /// of file names and subdirectory names found directly in the specified directory.
        /// </remarks>
        public static async Task<FolderInfo?> GetDirectoryInfoActivity(string directoryPath)
        {
            var rawResult = await Temporalio.Workflows.Workflow.ExecuteActivityAsync<object?>(
                AwaConstants.ActivityGetDirectoryInfo,
                new object?[] { directoryPath },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue,
                    StartToCloseTimeout = TimeSpan.FromSeconds(AwaConstants.FileIoTimeoutSeconds)
                });

            if (rawResult == null)
            {
                return null;
            }

            // Convert the raw result to FolderInfo
            return FolderInfo.FromJson(rawResult.ToString());
        }
    }
}

File: awa/sdk_dist/python/awa/client/utils/activity/get_directory_info_activity.py
from datetime import timedelta
from pathlib import Path

from temporalio import workflow

from awa.client import constants
from awa.client.models.folder_info import FolderInfo


async def get_directory_info_activity(directory_path: str | Path) -> FolderInfo:
    """Get information about a single directory including its immediate files and subdirectories.

    This function examines a single directory and returns information about its
    immediate contents without recursing into subdirectories. It provides lists
    of file names and subdirectory names found directly in the specified directory.

    Args:
        directory_path: The path to the directory to analyze. Can be a string or Path object.

    Returns:
        FolderInfo: A FolderInfo object containing:
            - path: The directory path
            - files: List of file names (not full paths)
            - subdirectories: List of subdirectory names (not full paths)

    """
    raw_result = await workflow.execute_activity(
        constants.ACTIVITY_GET_DIRECTORY_INFO,
        arg=str(directory_path),
        task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
        start_to_close_timeout=timedelta(seconds=constants.FILE_IO_TIMEOUT_SECONDS),
    )
    return FolderInfo(**raw_result)
